### ISSUE TITLE
Tree View Fails to load when complex `_id` data types are present

### ISSUE THREAD

**Issue by tnaum-ms** [microsoft/vscode-cosmosdb#2515]
When a dataset contains `_id` fields with more complex data types, the Tree View fails to render. This could be caused by:  
- Erroneous `_id` extraction logic.  
- Incorrect assumptions that `_id` is always of a specific type.  

**Task:**  
- Investigate the cause of the issue.  
- Implement a fix to handle complex `_id` data types correctly.  

**Data in the Table View:**

![Image](https://github.com/user-attachments/assets/ffd0ce0c-b54d-4537-9635-d16544e2140e)

**after switching to the Tree View:**

![Image](https://github.com/user-attachments/assets/2aaf69a8-d2e5-4d95-a5a0-150eeecb1eea)

**Comment by ghiscoding** [comment]
I was curious about this one, so I tried replicating this demo with just what I see in the print screen above. The issue might be caused by the fact that the SlickGrid DataView requires unique "id" and I'm not sure if your Id in this case is `_id`? If it is then multiple entries as objects instead of strings might react in different ways. In order to replicate a working demo, I added a formatter in the `_id` column to deal with the string or object data and the biggest change to get a working demo was to modify the data output by using a map to provide truly unique Ids as string. 

So here's the 2 modifications I've done

```ts
columnDefinitions = [
  {
        id: '_id', name: '_id', field: '_id', width: 220,
        filterable: true, sortable: true, exportWithFormatter: false,
        formatter: Formatters.multiple,
        params: {
          formatters: [
            // 1st formatter for ID
            ((_row, _cell, val) => {
              if (typeof val === 'object') {
                return JSON.stringify(val); // display stringified object
              }
              return val;
            }) as Formatter,
            // 2nd formatter for Tree Data icons
            Formatters.tree
          ]
        }
      },
      {
        id: 'type', name: 'type', field: '_id', minWidth: 90, filterable: true,
        formatter: (_row, _cell, val) => typeof val === 'string' ? 'ID' : '{}'
      },
];
```

and the use of map to add a unique Id as string
```ts
loadData() {
    const data = [
      { _id: '667591e', parentId: null },
      { _id: '67224b2', parentId: null },
      { _id: { mySpecialID: '223344' }, parentId: null },
      { _id: { mySpecialID: '424242' }, parentId: '667591e' },
    ];

    // make sure that the "id" is unique by using map
    const output = data.map((entry) => ({
      _id: entry._id,
      id: typeof entry._id === 'string' ? entry._id : entry._id['mySpecialID'],
      parentId: entry.parentId,
    }));

    console.log('data', output);
    return output;
  }
```

You can see a working Stackblitz of this change at this link (I modified the Slickgrid-React Example 27, so you'll need to route to that example to see it live)
https://stackblitz.com/edit/github-bnyowcpe?file=src%2Fexamples%2Fslickgrid%2FExample27.tsx

**Comment by tnaum-ms** [comment]
@ghiscoding Thank you for your feedback and ideas! I really appreciate your help. 🙏  

It took me a while to get back to this issue, but now that we have a **public schedule in place**, there’s no escaping it. 😉  

I remembered the requirement for unique IDs, so I actually generate one for every entry in the tree instead of deriving them from my documents—just to be safe.  

https://github.com/microsoft/vscode-cosmosdb/blob/dd5a02b699c1df5e651e8ba6863b47b6ccf064a2/src/utils/convertors.ts#L113-L115

The **core of the issue** was actually something else—I didn’t consider how a `null` value in the `_id` field would be represented, I assumed some sort of a bson object. Turns out, it’s an **actual** `null`. I hadn’t anticipated that, and that’s what was causing the issue. It didn’t notice it during the original test run, but thankfully, I still had that file for debugging.  

### Long story short:
- No issues with **SlickGrid** 🎉  
- No issues with **Tree View** 🎉  
- Just a rookie mistake with accessing a `null` value 🤦‍♂️

**Comment by sevoku** [comment]
@tnaum-ms let's don't close issues until the corresponding fix has been merged. GH will close this automatically once merged.

**Comment by tnaum-ms** [comment]
> [@tnaum-ms](https://github.com/tnaum-ms) let's don't close issues until the corresponding fix has been merged. GH will close this automatically once merged.

Ah, indeed, marking it as "done" in the project closed the issue here. I'll stick with moving it to "In Review" as we did in the past.
